src/common/src/block_builder.cc

```
@@ -0,0 +1,147 @@
+#include "block_builder.h"
+#include <bits/stdc++.h>
+
+BlockBuilder::BlockBuilder(void)
+{
+}
+
+BlockBuilder::BlockBuilder(const std::vector<std::vector<std::string> > &records)
+{
+    for (size_t i = 1; i < records.size(); ++i)
+    {
+        const auto&  record = records[i];
+        types::Block block  = ConvertRecordToBlock(record);
+        blocks_.push_back(block);
+    }
+}
+
+void AssignBlockInfrastructure(types::Block &block, const std::string &infrastructure)
+{
+    std::string infra = infrastructure;
+    std::transform(infra.begin(), infra.end(), infra.begin(), ::tolower);
+    if (infra.find("station") != std::string::npos)
+    {
+        block.has_station  = true;
+        block.station_name = infra;
+    }
+    if (infra.find("railway") != std::string::npos)
+    {
+        block.has_crossing = true;
+    }
+    if (infra.find("light") != std::string::npos)
+    {
+        block.has_light = true;
+    }
+    if (infra.find("switch") != std::string::npos)
+    {
+        block.has_switch = true;
+    }
+}
+
+types::Block BlockBuilder::ConvertRecordToBlock(const std::vector<std::string> &record)
+{
+    types::Block block;
+    block.section = record[1][0];
+    try {
+        block.block = std::stoi(record[2]);
+    }
+    catch (const std::invalid_argument& e) {
+        std::cerr << "Error: " << "Block: " << record[2] << " Invalid input for block number: " << record[2] << std::endl;
+        throw;
```

I think it would be better to return an error code than throw.  Feel free to add new generic error codes to the error enum in `types.h` or create your own enum of error codes specific to this module.  This will also allow you to use a series of `if` statements rather try/catch blocks.